### PR TITLE
Add NSMenuToolbarItem implementation

### PR DESCRIPTION
**Pull request by gcasa**
Add implementation of NSMenuToolbarItem.

### CODE REVIEW

**Comment by gcasa** [comment]
Hey @fredkiefer let me know if there are any issues with this.  Thanks. :)

**Comment by gcasa** [comment]
@fredkiefer This is a relatively simple change.   Please take a moment to review.

File: Source/NSMenuToolbarItem.m [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+*/
+
+#import "AppKit/NSMenuToolbarItem.h"
+#import "AppKit/NSMenu.h"
+#import "AppKit/NSImage.h"
+
+#import "GNUstepGUI/GSTheme.h"
+
+@interface NSToolbarItem (Private)
+- (NSView *) _backView;
```
Comment (fredkiefer) [inline Source/NSMenuToolbarItem.m:32]: You don‘t see to use this method. Is it a left over from a previous attempt implement this class? Maybe that other version was a better fit?

File: Source/NSMenuToolbarItem.m [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+- (void) setMenu: (NSMenu *)menu
+{
+  ASSIGN(_menu, menu);
+}
+
+- (void) _showMenu: (id)sender
+{
+  [[GSTheme theme] rightMouseDisplay: _menu
```
Comment (fredkiefer) [inline Source/NSMenuToolbarItem.m:86]: Where is this menu being displayed? Should we fake an event here to pass a location or are there better ways to control that?

**Review by fredkiefer — COMMENTED** [review]
Looks basically good but with two tiny questions.

File: Source/NSMenuToolbarItem.m (hunk 2 above)
Comment (gcasa) [reply Source/NSMenuToolbarItem.m:86]: The menu shows when you right-click on the NSToolbarItem icon that represents it.

File: Source/NSMenuToolbarItem.m (hunk 1 above)
Comment (gcasa) [reply Source/NSMenuToolbarItem.m:32]: This method is something I thought I would need but ended up not using.  I have removed it.

File: Source/NSMenuToolbarItem.m (hunk 2 above)
Comment (gcasa) [reply Source/NSMenuToolbarItem.m:86]: Here is an example on GNUstep:

menu_item_toolbar_test

And on the mac:

mac_menu_toolbar_item

Both are using the test code referenced in the previous comments.

**Comment by gcasa** [comment]
Sorry about the lateness of my reply, I haven't been feeling well for several days.

**Review by fredkiefer — APPROVED** [review]
Look great

**Pull request merged**